framework/Runners/SharedMemoryRunner.py

```diff
@@ -146,6 +146,7 @@ def kill(self):
       @ In, None
       @ Out, None
     """
-    self.raiseAWarning("Terminating "+self.thread.pid+ " Identifier " + self.identifier)
+    pid = self.thread.pid if self.thread is not None else None
+    self.raiseAWarning('Terminating job with PID "{}" and RAVEN identifier "{}"'.format(pid, self.identifier))
     os.kill(self.thread.pid, signal.SIGTERM)
```

Comment:
if it is none this should fail...right?

Reply:
added a caveat to prevent the "kill" call if it's not active.